### ISSUE TITLE
Stuck on 64KB+ data

### ISSUE THREAD

**Issue by mikkleini** [jchristn/WatsonTcp#42]
Hi Joel,

I hit one issue when transmitting data (messages) larger than TCP maximum (over 64KB) from server to client. It's in the same place as issue #31 . There's a while (true) loop in WatsonMessage.cs:

```
while (true)
{
	read = await _NetworkStream.ReadAsync(buffer, 0, buffer.Length);
	if (read == count)
	{
		ret = new byte[read];
		Buffer.BlockCopy(buffer, 0, ret, 0, read);
		break;
	}
	if (read == 0)
	{
		throw new SocketException();
	}
}
```

In my case server response (and therefore buffer.Length) is 256KB but sometimes the ReadAsync function returns less data (64KB). Basically it returns data in frames. In such case this loop stucks forever.

Please check this post, there's a fix and link to explanation:
https://stackoverflow.com/a/55755530/4370728

I haven't checked how if same issue is on server side also...

**Comment by jchristn** [comment]
Hi @mikkleini - testing this out now, thanks for mentioning this.  I need to maintain a counter of bytes remaining (set from the ```count``` parameter in ```Message.ReadFromNetwork``` and manage it that way instead.

**Comment by jchristn** [comment]
Thanks @mikkleini !  Merged in your changes and pushed v2.1.5 to NuGet!

NuGet: https://www.nuget.org/packages/WatsonTcp/2.1.5
Commit: https://github.com/jchristn/WatsonTcp/commit/c97a29fce905871ace5895f39ffdc8fcb9c55473
